packages/xpath/src/lib/geo/Geotrace.ts

```diff
@@ -0,0 +1,82 @@
+import type { Geopoint } from './Geopoint.ts';
+import { geopointCodec, type GeopointRuntimeValue } from './geopointCodec.ts';
+import { GeotraceLine } from './GeotraceLine.ts';
+
+export type GeotracePoints = readonly [Geopoint, Geopoint, ...Geopoint[]];
+
+const isGeotracePoints = (
+	geopoints: readonly GeopointRuntimeValue[]
+): geopoints is GeotracePoints => {
+	return (
+		geopoints.length >= 2 &&
+		geopoints.every((geopoint) => {
+			return geopoint != null;
+		})
+	);
+};
+
+const collectLines = (geopoints: GeotracePoints): readonly GeotraceLine[] => {
+	return geopoints.reduce((acc, geopoint, i) => {
+		if (i === 0) {
+			return acc;
+		}
+
+		// Non-null assertion safe: we ensure at least 2 points, and skip index 0.
+		const start = geopoints[i - 1]!;
+		const end = geopoint;
+
+		acc.push(
+			new GeotraceLine({
+				start,
+				end,
+			})
+		);
+
+		return acc;
+	}, Array<GeotraceLine>());
+};
+
+export class Geotrace {
+	static fromEncodedGeotrace(encoded: string): Geotrace | null {
+		const geopoints = encoded
+			.replace(/\s*;\s*$/, '')
```

Comment:
This is _still_ inconsistent with JavaRosa! After playing around with it a bit more, it turns out we should ignore any number of trailing semicolons, not just one!

I did a quick search for the actual code handling this detail, but I don't see it. I strongly suspect there's some **Java nuance** at play, but I couldn't say for sure at my current familiarity with the language.

Reply:
Maybe something like `.replace(/^[\s;]+|[\s;]+$/g, '')`

But it'll remove spaces and semicolon at the beginning and end of the string, it might need a different one for the empty middle sections